front/src/modules/ui/input/relation-picker/hooks/useEntitySelectScroll.ts

```diff
@@ -60,38 +64,36 @@ export const useEntitySelectScroll = <
       }
     },
     RelationPickerHotkeyScope.RelationPicker,
-    [setRelationPickerHoverIndex, entities],
+    [hoverIds],
   );
 
   useScopedHotkeys(
     Key.ArrowDown,
     () => {
-      setRelationPickerHoverIndex((prevSelectedIndex) =>
-        Math.min(prevSelectedIndex + 1, (entities?.length ?? 0) - 1),
-      );
-
-      const currentHoveredRef = containerRef.current?.children[
-        relationPickerHoverIndex
-      ] as HTMLElement;
+      const nextId = hoverIds[nextIndex];
```

Comment:
Please always try to keep a maximum of context in your variable naming, avoid generic names and also shortened names.

Here it should be at something like `nextHoverableId` to tell that we're using this variable for guessing what will be the next item in the list.